tests/v2/validation/provisioning/rke2/etcd_only_backup_restore_test.go

```diff
@@ -0,0 +1,305 @@
+	initialK8sVersion := r.kubernetesVersions[0]
+	logrus.Infof("running etcd snapshot restore test.............")
+	subSession := r.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := r.client.WithSession(subSession)
+	require.NoError(r.T(), err)
+
+	logrus.Infof("creating kube provisioning client.............")
+	kubeProvisioningClient, err := r.client.GetKubeAPIProvisioningClient()
+	require.NoError(r.T(), err)
+	logrus.Infof("kube provisioning client created.............")
+
+	clusterName := provisioning.AppendRandomString(provider.Name)
+
+	logrus.Infof("creating rke2Cluster.............")
+	clusterResp, err := createRKE2NodeDriverCluster(client, provider, clusterName, initialK8sVersion, r.ns, r.cnis[0])
+	require.NoError(r.T(), err)
+	require.Equal(r.T(), clusterName, clusterResp.ObjectMeta.Name)
+	logrus.Infof("rke2Cluster create request successful.............")
+
+	logrus.Infof("creating watch over cluster.............")
+	r.watchAndWaitForCluster(kubeProvisioningClient, clusterName)
+	logrus.Infof("cluster is up and running.............")
+
+	// Get clusterID by clusterName
+	logrus.Info("getting cluster id.............")
+	clusterID, err := clusters.GetClusterIDByName(client, clusterName)
+	require.NoError(r.T(), err)
+	logrus.Info("got cluster id.............", clusterID)
+
+	logrus.Info("getting local cluster id.............")
+	localClusterID, err := clusters.GetClusterIDByName(client, localClusterName)
+	require.NoError(r.T(), err)
+	logrus.Info("got local cluster id.............", localClusterID)
+
+	logrus.Infof("creating watch over pods.............")
+	r.watchAndWaitForPods(client, clusterID)
+	logrus.Infof("All pods are up and running.............")
+
+	// creating the workload W1
+	logrus.Infof("creating a workload(nginx deployment).............")
+
+	wloadBeforeRestoreLabels := map[string]string{}
+	wloadBeforeRestoreLabels["workload.user.cattle.io/workloadselector"] = fmt.Sprintf("apps.deployment-%v-%v", r.ns, wloadBeforeRestore)
+
+	containerTemplate := workloads.NewContainer("ngnix", "nginx", v1.PullAlways, []v1.VolumeMount{}, []v1.EnvFromSource{})
+	podTemplate := workloads.NewPodTemplate([]v1.Container{containerTemplate}, []v1.Volume{}, []v1.LocalObjectReference{}, wloadBeforeRestoreLabels)
+	deploymentBeforeBackup := workloads.NewDeploymentTemplate(wloadBeforeRestore, r.ns, podTemplate, wloadBeforeRestoreLabels)
+
+	// creating steve client
+	steveclient, err := client.Steve.ProxyDownstream(clusterID)
+	require.NoError(r.T(), err)
+
+	deploymentResp, err := createDeployment(deploymentBeforeBackup, steveclient, client, clusterID)
+	require.NoError(r.T(), err)
+	require.Equal(r.T(), deploymentBeforeBackup.Name, deploymentResp.ObjectMeta.Name)
+	logrus.Infof("%v is ready.............", deploymentBeforeBackup.Name)
+
+	// creating the ingress1
+	logrus.Infof("creating an ingress.............")
+
+	exactPath := networkingv1.PathTypeExact
+	paths := []networkingv1.HTTPIngressPath{
```

Comment:
Can be this abstracted into a function defined in `etcd_backup_restore.go`? Maybe something called `CreateIngressPath`, just to hide the implementation in the actual test.